src/lib.rs

```diff
@@ -2180,11 +2210,17 @@ impl Build {
             ArchSpec::Catalyst(_) => "macosx".to_owned(),
         };
 
-        self.print(&format!("Detecting {} SDK path for {}", os, sdk));
-        let sdk_path = self.apple_sdk_root(sdk.as_str())?;
-        cmd.args.push("-isysroot".into());
-        cmd.args.push(sdk_path);
-        cmd.args.push("-fembed-bitcode".into());
+        // AppleClang sometimes needs sysroot even for darwin
+        if cmd.is_xctoolchain_clang() || !target.ends_with("-darwin") {
+            self.print(&format!("Detecting {} SDK path for {}", os, sdk));
+            let sdk_path = self.apple_sdk_root(sdk.as_str())?;
+            cmd.args.push("-isysroot".into());
+            cmd.args.push(sdk_path);
+        }
+
+        if !is_mac {
```

Comment:
Can you explain why this is conditional now?

Reply:
The whole function `apple_flags` - which was previously `ios_watchos_flags` was not called from macos target before. I'd like to keep the previous behavior for macos target in this PR. If you'd like to add this to macos too, I will remove it.

Reply:
Thanks, good catch.